src/firefly/js/tables/ui/BasicTableView.jsx

```diff
@@ -0,0 +1,233 @@
+/*
+ * License information at https://github.com/Caltech-IPAC/firefly/blob/master/License.txt
+ */
+
+import React, {PropTypes} from 'react';
+import sCompare from 'react-addons-shallow-compare';
+import FixedDataTable from 'fixed-data-table';
+import Resizable from 'react-component-resizable';
+import {debounce, get, isEmpty, pick} from 'lodash';
+
+import {SelectInfo} from '../SelectInfo.js';
+import {FilterInfo} from '../FilterInfo.js';
+import {SortInfo} from '../SortInfo';
+import {tableToText} from '../TableUtil.js';
+import {TextCell, HeaderCell} from './TableRenderer.js';
+
+import './TablePanel.css';
+
+const {Table, Column} = FixedDataTable;
+
+export class BasicTableView extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            showMask: false,
+            widthPx: 75,
+            heightPx: 75,
+            columnWidths: makeColWidth(props.columns, props.data, props.showUnits)
+        };
+
+        this.onResize =  debounce((size) => {
+                if (size) {
+                    var widthPx = size.width;
+                    var heightPx = size.height;
+                    this.setState({widthPx, heightPx});
+                }
+            }, 100);
+
+        this.onColumnResizeEndCallback = this.onColumnResizeEndCallback.bind(this);
+        this.rowClassName = this.rowClassName.bind(this);
+    }
+
+    onColumnResizeEndCallback(newColumnWidth, columnKey) {
+        var columnWidths = Object.assign({}, this.state.columnWidths, {[columnKey]: newColumnWidth});
+        this.setState({columnWidths});
+    }
+
+    rowClassName(index) {
+        const {hlRowIdx} = this.props;
+        return (hlRowIdx === index) ? 'tablePanel__Row_highlighted' : '';
+    }
+
+    componentWillReceiveProps(nProps) {
+        if (isEmpty(this.state.columnWidths) && !isEmpty(nProps.columns)) {
+            this.setState({columnWidths: makeColWidth(nProps.columns, nProps.data, nProps.showUnits)});
+        }
+        this.setState({showMask: false});
+    }
+
+    shouldComponentUpdate(nProps, nState) {
+        return sCompare(this, nProps, nState);
+    }
+
+    render() {
+        const {columns, data, hlRowIdx, showUnits, showFilters, filterInfo,
+                    sortInfo, callbacks, textView, rowHeight} = this.props;
+        const {widthPx, heightPx, columnWidths, showMask} = this.state;
+
+        if (isEmpty(columns)) return (<div style={{top: 0}} className='loading-mask'/>);
```

Comment:
You are right.  We should adhere to HTML5 which is closing any non-void elements tags.  I'll fix it.